source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -519,18 +545,42 @@ ShaderNode* ShaderGraph::addNode(const Node& node, ShaderGenerator& shadergen)
         }
     }
 
+    const string& targetColorSpace = _document ? _document->getAttribute(Element::COLOR_SPACE_ATTRIBUTE) : EMPTY_STRING;
```

Comment:
This would work assuming implementations have been loaded already (which I presume should be the case)

It would still be nice if you can skip all CMS if the language is not supported. Maybe something like:
```
bool ColorManagementSystem::supportsLanguage(const string& language)
```

Reply:
Not sure about the `supportsLanguage()` method? Where would that be used?

There is already an error thrown if the implementation for a transform is not found, which also includes having the right language.

Reply:
I'm currently using it on line 554 of ShaderGraph.cpp to early out.

Reply:
This should have been to test if the CMS supports the shadergen language -- the other test is of course of no use :)